src/triangle.rs

```diff
@@ -0,0 +1,655 @@
+    pub fn with_distances(self, distances: [f64; 3]) -> Self {
+        Self::from_centroid_distances(self.centroid(), distances)
+    }
+
+    /// The centroid of the [`Triangle`]
+    #[inline]
+    pub fn centroid(&self) -> Point {
+        Point::new(
+            (self.a.x + self.b.x + self.c.x) / 3.0,
+            (self.a.y + self.b.y + self.c.y) / 3.0,
+        )
+    }
+
+    /// The euclidean distance of each vertex from the centroid
+    #[inline]
+    pub fn distance(&self) -> [f64; 3] {
+        let centroid = self.centroid();
+
+        [
+            self.a.distance(centroid),
+            self.b.distance(centroid),
+            self.c.distance(centroid),
+        ]
+    }
+
+    /// The offset of each vertex from the centroid
+    #[inline]
+    pub fn sizes(&self) -> [Size; 3] {
+        let centroid = self.centroid().to_vec2();
+
+        [
+            (self.a.to_vec2() - centroid).to_size(),
+            (self.b.to_vec2() - centroid).to_size(),
+            (self.c.to_vec2() - centroid).to_size(),
+        ]
+    }
+
+    /// The perimeter of the [`Triangle`]
+    #[inline]
+    fn perimeter(&self) -> f64 {
+        self.a.distance(self.b) + self.b.distance(self.c) + self.c.distance(self.a)
+    }
+
+    /// The area of the [`Triangle`]
+    #[inline]
+    pub fn area(&self) -> f64 {
+        let ab = self.a.distance(self.b);
+        let ac = self.a.distance(self.c);
+        let bc = self.b.distance(self.c);
+
+        // cos rule
+        let theta = ((bc * bc - ab * ab - ac * ac) / (-2.0 * ab * ac)).acos();
+
+        // A = 1/2*a*b*sin(C)
+        (ab * ac * theta.sin()) / 2.0
+    }
+
+    // TODO: maybe make height functions for right and non right angled triangles
+
+    /// The area of a right angled [`Triangle`]
+    ///
+    /// NOTE: assumed [`Triangle::organise`]d positions
+    #[inline]
+    pub fn right_angled_area(&self) -> f64 {
+        // A = 1/2*b*h
+        (self.b.distance(self.c) * (self.a.y - self.b.y)) / 2.0
+    }
+
+    /// Updates [`Triangle`]'s vertice positions
+    /// such that [`Triangle::a`] is topmost, [`Triangle::b`] is leftmost, and [`Triangle::c`] is rightmost
+    #[inline]
+    pub fn organise(self) -> Self {
+        let mut points = self.as_array();
+        points.sort_by(|x, y| x.rightmost(y));
+
+        let a = self.topmost();
+        let b = *points.iter().find(|&&p| p != a).unwrap();
+        let c = *points.iter().rev().find(|&&p| p != a).unwrap();
+        Self::new(a, b, c)
+    }
+
+    /// Vertex coordinate ([`Point`]) of [`Triangle`] with `x` ordinate
+    #[inline]
+    pub fn vertex_from_x(&self, x: f64) -> Option<Point> {
+        self.as_array().iter().find(|&v| v.x == x).copied()
+    }
+
+    /// Vertex coordinate ([`Point`]) of [`Triangle`] with `y` ordinate
+    #[inline]
+    pub fn vertex_from_y(&self, y: f64) -> Option<Point> {
+        self.as_array().iter().find(|&v| v.y == y).copied()
+    }
+
+    /// The topmost vertex of `self` as a [`Point`]
```

Comment:
I don't think these are justified, I think it's fairly specialized and users can write it themselves. In addition, there's a naming problem, as "topmost" implies a y-down coordinate system, and one of the conceits of kurbo is that it's agnostic to whether the coordinates are y-up or y-down.

Reply:
Oh right, their useless now anyway because the `organise` is useless